spec/controllers/api/v1/reviews_controller_spec.rb

```diff
@@ -0,0 +1,79 @@
+require "rails_helper"
+
+RSpec.describe Api::V1::ReviewsController, type: :controller do
+
+  let!(:test_region) { Region.create(
+    name: "Test Region 1",
+    image_url: "https://www.google.com",
+  ) }
+
+  let!(:test_flavor) { Flavor.create(
+    name: "Chocolatey Goodness",
+    description: "It is brown.",
+    image_url: "www.chocolate.com",
+    region: test_region
+  )}
+
+  let!(:test_user) { User.create(
+    username: "Bob",
+    email: "bob@robert.com",
+    password: "something"
+  )}
+
+  describe "POST#create" do
+
+    let!(:good_review) {{ review: {
+      overall: 3,
+      sweetness: 3,
+      mouth_feel: 3,
+      taste: 3,
+      comment: "Meh.",
+      manufacturer: "Not worth saying."
+    },
+      flavor_id: test_flavor.id
+    }}
+
+    it "should not persist if the user is not authenticated" do
+      count = Review.count
+
+      post :create, params: good_review
+      returned_json = JSON.parse(response.body)
+
+      expect(Review.count).to eq(count)
+      expect(returned_json["notice"]).to eq("Please login before submitting a review.")
+    end
+
+    it "should persist if the review is valid" do
+      count = Review.count
+
+      sign_in test_user
+      post :create, params: good_review
+      returned_json = JSON.parse(response.body)
+
+      expect(Review.count).to eq(count + 1)
+      expect(returned_json["notice"]).to eq("Review submitted successfully!")
+    end
+
+    let!(:bad_review) {{ review: {
+      overall: 5,
+      sweetness: 3,
+      mouth_feel: 4,
+      taste: 5,
+      comment: "",
+      manufacturer: ""
+    },
+      flavor_id: test_flavor.id
+    }}
```

Comment:
the indentation on these is kinda funky. this would be better. Not sure if you can remove the extra curly brace as well potentially
```
{  
  flavor_id: .... ,
  review: {
      overall: 5,
      sweetness: 3,
      mouth_feel: 4,
      taste: 5,
      comment: "",
      manufacturer: ""
     }
}
```

Reply:
Formatting updated, we confirmed Rspec still passed with the changes as well.